iOSDeviceManager/Utilities/DeviceUtils.m

```diff
@@ -140,25 +140,25 @@ + (NSString *)defaultSimulator {
     int major = XcodeUtils.versionMajor + 2;
     int minor = XcodeUtils.versionMinor;
 
-    if (XcodeUtils.versionMajor == 10) {
+    NSString *deviceVersion;
+
+    if (XcodeUtils.versionMajor == 11) {
+        deviceVersion = @"Xs";
+    } else if (XcodeUtils.versionMajor == 10) {
         if (XcodeUtils.versionMinor < 2) {
-            return [NSString
-                    stringWithFormat:@"iPhone XS (%d.%d)",
-                    major,
-                    minor];
+            deviceVersion = @"XS";
         } else {
-            return [NSString
-                    stringWithFormat:@"iPhone Xs (%d.%d)",
-                    major,
-                    minor];
+            deviceVersion = @"Xs";
         }
     } else {
-        return [NSString
-                stringWithFormat:@"iPhone %d (%d.%d)",
-                XcodeUtils.versionMajor - 1,
+        deviceVersion = [NSString stringWithFormat: @"%d", XcodeUtils.versionMajor - 1];
+    }
+
+    return [NSString
+                stringWithFormat:@"iPhone %@ (%d.%d)",
+                deviceVersion,
                 major,
                 minor];
-    }
```

Comment:
Can you add a test for this to DeviceUtilsTest.m?

Use OCMock to stub the return values from XcodeUtils.

If you have time, the XcodeUtils should be rewritten to return `NSUInteger` instead of `int`.

Reply:
Sure.
We have added tests for `DeviceUtils.defaultSimulator` method and reworked `XcodeUtils` implementation in order to use `NSUInteger` instead of `int`.